elkeschaper/tral
# Infrastructure improvements

This includes a series of changes aimed at improving TRAL testing, installation, and code quality.

- Testing
  - Add tox for testing in multiple environments
  - Add travis CI server
    - Currently doesn't work due to #29
  - `python setup.py test` runs tests (pytest)
- Installation
  - Specify data directory location with `--tral-home` when running `setup.py`. This is still only useful for installing with sudo, since TRAL only checks in the home directory at runtime (see #9).
  - clean up setup.py
  - Specify developer requirements as 'extras'. For instance, `python setup.py install .[docs]` to build documentation, or `pip install -r requirements_dev.txt` for everything
- Code quality
  - Add flake8 tox target
  - Fix most flake8 problems. Some additional problems are obscured by `ignore` settings in `.flake8`. These should be fixed and not ignored.
